### PR TITLE
Implement alert/confirm through Chrome's dialog manager

### LINKED ISSUE #2644: No alert or confirm dialogs in iframes
Currently there's no support for `alert` and `confirm` dialogs in iframes (see discussion in #2522).

Once the "Session wide preload script" #2605 is ready, I think we should implement a default preload that will add the missing dialogs (kind of like what override.coffee does for the parent window).

### PR DESCRIPTION
**Pull request by kevinsawicki**
This allows `window.alert` and `window.confirm` to work from `<iframe>` tags and sandboxed windows.

Refs #2644

### CODE REVIEW

File: atom/browser/atom_javascript_dialog_manager.cc [hunk 1]
```diff
@@ -5,20 +5,50 @@
...
+
+  if (message_type != JavaScriptMessageType::JAVASCRIPT_MESSAGE_TYPE_ALERT &&
+      message_type != JavaScriptMessageType::JAVASCRIPT_MESSAGE_TYPE_CONFIRM) {
+    callback.Run(false, base::string16());
+    return;
+  }
+
+  std::vector<std::string> buttons;
+  if (message_type == JavaScriptMessageType::JAVASCRIPT_MESSAGE_TYPE_CONFIRM) {
+    buttons.push_back("OK");
```
Comment (deepak1556) [inline atom/browser/atom_javascript_dialog_manager.cc]: "OK" should be present for both confirm and alert types.

**Comment by kevinsawicki** [comment]
> The overrides in lib/renderer/override,js can be removed ?

I didn't remove them because they both support a second `title` parameter that isn't part of the DOM API and so it isn't supported via the Chrome dialog manager. And so I didn't want to break any existing behavior where apps are passing a custom title.

We could remove them in 2.0 though.

**Comment by deepak1556** [comment]
Makes sense, thanks for the explanation :+1:

**Comment by jackywang863** [comment]
in newest master sources we did not found the above commits, so  when commit the 3 file into master?

**Pull request merged**